sdk/nodejs/docker.ts

```diff
@@ -260,18 +260,15 @@ async function buildAndPushImageWorkerAsync(
     }
 
     // If the container specified a cacheFrom parameter, first set up the cached stages.
-    let cacheFrom = Promise.resolve<string[] | undefined>(undefined);
+    let cacheFrom: string[] = [];
```

Comment:
first, passing around 'undefined' just made later code more complex.  We can instead just pass around an empty array.  The later code already checks that the array is non-empty before proceeding.